src/jl_uv.c

```diff
@@ -830,6 +830,56 @@ JL_DLLEXPORT int jl_tty_set_mode(uv_tty_t *handle, int mode)
     return uv_tty_set_mode(handle, mode);
 }
 
+typedef int (*work_cb_t)(void *, void *);
+typedef void (*notify_cb_t)(int);
+
+struct work_baton {
+    uv_work_t req;
+    work_cb_t work_func;
+    void      *work_args;
+    void      *work_retval;
+    notify_cb_t notify_func;
+    pid_t     tid;
+    int       notify_idx;
+};
+
+#ifdef _OS_LINUX_
+#include <sys/syscall.h>
+#endif
+
+void jl_work_wrapper(uv_work_t *req) {
+    struct work_baton *baton = (struct work_baton*) req->data;
+#ifdef _OS_LINUX_
+    baton->tid = syscall(SYS_gettid);
```

Comment:
`gettid` does not have a glibc wrapper. `syscall` is the documented way to use it. That said, we don't need the thread id, I had added it in there for debugging purposes - to ensure that the calls are actually being executed on different threads.